streaming/index.js

```diff
@@ -25,15 +29,86 @@ dotenv.config({
   path: path.resolve(__dirname, path.join('..', dotenvFile))
 });
 
-log.level = process.env.LOG_LEVEL || 'verbose';
+const logger = pino({
+  name: "streaming",
+  // Reformat the log level to a string:
+  formatters: {
+    level: (label) => {
+      return {
+        level: label
+      };
+    },
+  },
+  redact: {
+    paths: [
+      'req.headers["sec-websocket-key"]',
+      // Note: we currently pass the AccessToken via the websocket subprotocol
+      // field, an anti-pattern, but this ensures it doesn't end up in logs.
+      'req.headers["sec-websocket-protocol"]',
+      'req.headers.authorization',
+      'req.headers.cookie',
+      'req.query.access_token'
+    ]
+  }
+});
+
+/**
+ * Generates the Request ID for logging and setting on responses
+ * @param {http.IncomingMessage} req
+ * @param {http.ServerResponse} [res]
+ * @returns {import("pino-http").ReqId}
+ */
+function genReqId(req, res) {
+  if (req.id) {
+    return req.id;
+  }
+
+  req.id = uuid.v4();
+
+  // Allow for usage with WebSockets:
+  if (res) {
+    res.setHeader('X-Request-Id', req.id);
+  }
+
+  return req.id;
+}
+
+/**
+ * Request log sanitizer to prevent logging access tokens in URLs
+ * @param {http.IncomingMessage} req
+ */
+const sanitizeReqLog = (req) => {
+  const log = pinoHttpSerializers.req(req);
+  if (typeof log.url === 'string') {
+    log.url = log.url.replace(/(access_token)=([A-Z0-9a-z_]+)/gi, '$1=[Redacted]');
+  }
+  return log;
+};
+
+const httpLogger = pinoHttp({
+  logger,
+  genReqId,
+  serializers: {
+    req: sanitizeReqLog
+  }
+});
+
+if (process.env.LOG_LEVEL && Object.keys(logger.levels.values).includes(process.env.LOG_LEVEL)) {
+  logger.level = process.env.LOG_LEVEL;
+} else if (environment === 'development') {
+  logger.level = 'debug';
+} else {
+  logger.level = 'info';
+}
```

Comment:
I am tempted to move this into a generic `setupLogging` function that takes in the environment, and returns a `logger` and `httpLogger` back and also exposes the `sanitizeRequestForLogging` (`sanitizeReqLog`) method.

Would appreciate thoughts on if this makes sense as a change?